File: CS/Form1.cs
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;
using System.Data;

namespace CreateEditItem
{
	/// <summary>
	/// Summary description for Form1.
	/// </summary>
	public class Form1 : System.Windows.Forms.Form
	{
		private DevExpress.XtraBars.BarManager barManager1;
		private DevExpress.XtraBars.BarDockControl barDockControlTop;
		private DevExpress.XtraBars.BarDockControl barDockControlBottom;
		private DevExpress.XtraBars.BarDockControl barDockControlLeft;
		private DevExpress.XtraBars.BarDockControl barDockControlRight;
		private DevExpress.XtraBars.Bar barStandard;
		private DevExpress.XtraBars.Bar barStatus;
		/// <summary>
		/// Required designer variable.
		/// </summary>
		private System.ComponentModel.Container components = null;

		public Form1()
		{
			//
			// Required for Windows Form Designer support
			//
			InitializeComponent();

			//
			// TODO: Add any constructor code after InitializeComponent call
			//
		}

		/// <summary>
		/// Clean up any resources being used.
		/// </summary>
		protected override void Dispose( bool disposing )
		{
			if( disposing )
			{
				if (components != null) 
				{
					components.Dispose();
				}
			}
			base.Dispose( disposing );
		}

		#region Windows Form Designer generated code
		/// <summary>
		/// Required method for Designer support - do not modify
		/// the contents of this method with the code editor.
		/// </summary>
		private void InitializeComponent()
		{
			this.barManager1 = new DevExpress.XtraBars.BarManager();
			this.barStandard = new DevExpress.XtraBars.Bar();
			this.barStatus = new DevExpress.XtraBars.Bar();
			this.barDockControlTop = new DevExpress.XtraBars.BarDockControl();
			this.barDockControlBottom = new DevExpress.XtraBars.BarDockControl();
			this.barDockControlLeft = new DevExpress.XtraBars.BarDockControl();
			this.barDockControlRight = new DevExpress.XtraBars.BarDockControl();
			((System.ComponentModel.ISupportInitialize)(this.barManager1)).BeginInit();
			this.SuspendLayout();
			// 
			// barManager1
			// 
			this.barManager1.Bars.AddRange(new DevExpress.XtraBars.Bar[] {
																			 this.barStandard,
																			 this.barStatus});
			this.barManager1.DockControls.Add(this.barDockControlTop);
			this.barManager1.DockControls.Add(this.barDockControlBottom);
			this.barManager1.DockControls.Add(this.barDockControlLeft);
			this.barManager1.DockControls.Add(this.barDockControlRight);
			this.barManager1.Form = this;
			this.barManager1.MaxItemId = 0;
			this.barManager1.StatusBar = this.barStatus;
			// 
			// barStandard
			// 
			this.barStandard.BarName = "Standard";
			this.barStandard.DockCol = 0;
			this.barStandard.DockRow = 0;
			this.barStandard.DockStyle = DevExpress.XtraBars.BarDockStyle.Top;
			this.barStandard.Text = "Standard";
			// 
			// barStatus
			// 
			this.barStatus.BarName = "Status bar";
			this.barStatus.CanDockStyle = DevExpress.XtraBars.BarCanDockStyle.Bottom;
			this.barStatus.DockCol = 0;
			this.barStatus.DockRow = 0;
			this.barStatus.DockStyle = DevExpress.XtraBars.BarDockStyle.Bottom;
			this.barStatus.OptionsBar.AllowQuickCustomization = false;
			this.barStatus.OptionsBar.DrawDragBorder = false;
			this.barStatus.OptionsBar.UseWholeRow = true;
			this.barStatus.Text = "Status bar";
			// 
			// Form1
			// 
			this.AutoScaleBaseSize = new System.Drawing.Size(5, 13);
			this.ClientSize = new System.Drawing.Size(296, 178);
			this.Controls.Add(this.barDockControlLeft);
			this.Controls.Add(this.barDockControlRight);
			this.Controls.Add(this.barDockControlBottom);
			this.Controls.Add(this.barDockControlTop);
			this.Name = "Form1";
			this.Text = "Form1";
			this.Load += new System.EventHandler(this.Form1_Load);
			((System.ComponentModel.ISupportInitialize)(this.barManager1)).EndInit();
			this.ResumeLayout(false);

		}
		#endregion

		/// <summary>
		/// The main entry point for the application.
		/// </summary>
		[STAThread]
		static void Main() 
		{
			Application.Run(new Form1());
		}

		private void Form1_Load(object sender, System.EventArgs e) {
			DevExpress.XtraBars.BarEditItem item;
			DevExpress.XtraEditors.Repository.RepositoryItemComboBox combo;
			DevExpress.XtraEditors.Repository.RepositoryItemProgressBar progress;

			barManager1.ForceInitialize();

			// create combo
			combo = barManager1.RepositoryItems.Add("ComboBoxEdit") as DevExpress.XtraEditors.Repository.RepositoryItemComboBox;
			combo.Items.Add("Item A");
			combo.Items.Add("Item B");
			item = new DevExpress.XtraBars.BarEditItem(barManager1);
			item.Edit = combo;
			item.EditValue = "[n/a]";
			item.Width = 100;
			barStandard.AddItem(item);

			// create progress
			progress = barManager1.RepositoryItems.Add("ProgressBarControl") as DevExpress.XtraEditors.Repository.RepositoryItemProgressBar;
			item = new DevExpress.XtraBars.BarEditItem(barManager1);
			item.Edit = progress;
			item.EditValue = 26; // 26%
			item.Width = 150;
			barStatus.AddItem(item);
		}
	}
}

File: VB/Form1.vb
Imports System
Imports System.Drawing
Imports System.ComponentModel
Imports System.Windows.Forms

Namespace CreateEditItem

    ''' <summary>
    ''' Summary description for Form1.
    ''' </summary>
    Public Class Form1
        Inherits Form

        Private barManager1 As DevExpress.XtraBars.BarManager

        Private barDockControlTop As DevExpress.XtraBars.BarDockControl

        Private barDockControlBottom As DevExpress.XtraBars.BarDockControl

        Private barDockControlLeft As DevExpress.XtraBars.BarDockControl

        Private barDockControlRight As DevExpress.XtraBars.BarDockControl

        Private barStandard As DevExpress.XtraBars.Bar

        Private barStatus As DevExpress.XtraBars.Bar

        ''' <summary>
        ''' Required designer variable.
        ''' </summary>
        Private components As System.ComponentModel.Container = Nothing

        Public Sub New()
            '
            ' Required for Windows Form Designer support
            '
            InitializeComponent()
        '
        ' TODO: Add any constructor code after InitializeComponent call
        '
        End Sub

        ''' <summary>
        ''' Clean up any resources being used.
        ''' </summary>
        Protected Overrides Sub Dispose(ByVal disposing As Boolean)
            If disposing Then
                If components IsNot Nothing Then
                    components.Dispose()
                End If
            End If

            MyBase.Dispose(disposing)
        End Sub

#Region "Windows Form Designer generated code"
        ''' <summary>
        ''' Required method for Designer support - do not modify
        ''' the contents of this method with the code editor.
        ''' </summary>
        Private Sub InitializeComponent()
            barManager1 = New DevExpress.XtraBars.BarManager()
            barStandard = New DevExpress.XtraBars.Bar()
            barStatus = New DevExpress.XtraBars.Bar()
            barDockControlTop = New DevExpress.XtraBars.BarDockControl()
            barDockControlBottom = New DevExpress.XtraBars.BarDockControl()
            barDockControlLeft = New DevExpress.XtraBars.BarDockControl()
            barDockControlRight = New DevExpress.XtraBars.BarDockControl()
            CType(barManager1, System.ComponentModel.ISupportInitialize).BeginInit()
            Me.SuspendLayout()
            ' 
            ' barManager1
            ' 
            barManager1.Bars.AddRange(New DevExpress.XtraBars.Bar() {barStandard, barStatus})
            barManager1.DockControls.Add(barDockControlTop)
            barManager1.DockControls.Add(barDockControlBottom)
            barManager1.DockControls.Add(barDockControlLeft)
            barManager1.DockControls.Add(barDockControlRight)
            barManager1.Form = Me
            barManager1.MaxItemId = 0
            barManager1.StatusBar = barStatus
            ' 
            ' barStandard
            ' 
            barStandard.BarName = "Standard"
            barStandard.DockCol = 0
            barStandard.DockRow = 0
            barStandard.DockStyle = DevExpress.XtraBars.BarDockStyle.Top
            barStandard.Text = "Standard"
            ' 
            ' barStatus
            ' 
            barStatus.BarName = "Status bar"
            barStatus.CanDockStyle = DevExpress.XtraBars.BarCanDockStyle.Bottom
            barStatus.DockCol = 0
            barStatus.DockRow = 0
            barStatus.DockStyle = DevExpress.XtraBars.BarDockStyle.Bottom
            barStatus.OptionsBar.AllowQuickCustomization = False
            barStatus.OptionsBar.DrawDragBorder = False
            barStatus.OptionsBar.UseWholeRow = True
            barStatus.Text = "Status bar"
            ' 
            ' Form1
            ' 
            AutoScaleBaseSize = New System.Drawing.Size(5, 13)
            ClientSize = New System.Drawing.Size(296, 178)
            Me.Controls.Add(barDockControlLeft)
            Me.Controls.Add(barDockControlRight)
            Me.Controls.Add(barDockControlBottom)
            Me.Controls.Add(barDockControlTop)
            Name = "Form1"
            Text = "Form1"
            AddHandler Load, New EventHandler(AddressOf Form1_Load)
            CType(barManager1, System.ComponentModel.ISupportInitialize).EndInit()
            Me.ResumeLayout(False)
        End Sub

#End Region
        ''' <summary>
        ''' The main entry point for the application.
        ''' </summary>
        <STAThread>
        Shared Sub Main()
            Call Application.Run(New Form1())
        End Sub

        Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs)
            Dim item As DevExpress.XtraBars.BarEditItem
            Dim combo As DevExpress.XtraEditors.Repository.RepositoryItemComboBox
            Dim progress As DevExpress.XtraEditors.Repository.RepositoryItemProgressBar
            barManager1.ForceInitialize()
            ' create combo
            combo = TryCast(barManager1.RepositoryItems.Add("ComboBoxEdit"), DevExpress.XtraEditors.Repository.RepositoryItemComboBox)
            combo.Items.Add("Item A")
            combo.Items.Add("Item B")
            item = New DevExpress.XtraBars.BarEditItem(barManager1)
            item.Edit = combo
            item.EditValue = "[n/a]"
            item.Width = 100
            barStandard.AddItem(item)
            ' create progress
            progress = TryCast(barManager1.RepositoryItems.Add("ProgressBarControl"), DevExpress.XtraEditors.Repository.RepositoryItemProgressBar)
            item = New DevExpress.XtraBars.BarEditItem(barManager1)
            item.Edit = progress
            item.EditValue = 26 ' 26%
            item.Width = 150
            barStatus.AddItem(item)
        End Sub
    End Class
End Namespace
